feat: asset canister permissions can be set via upgrade args (#2237)
dfinity/portal: docs/developer-docs/integrations/sns/managing/sns-asset-canister.md

## Deploying an asset canister

The general overview of deploying an asset canister during an SNS launch is as follows:
- First, the asset canister must be created with or upgraded to a Wasm file from [dfx 0.15.2+](https://github.com/dfinity/sdk/blob/release-0.15.2/src/distributed/assetstorage.wasm.gz).
- Then, developers should use `revoke_permission` to remove their own permissions (especially `Commit`) that allow them to update the assets arbitrarily.

> - Then, after launching the SNS, the SNS's function should be registered to commit proposed changes.

Reviewer: Could we maybe link here to generic proposals? I wonder whether otherwise it is clear what "the SNS's function" refers to?

> - Last, using upgrade arguments, updated permissions can be set. This is only possible after launching an SNS because before the launch the principal id of SNS governance is not known yet

Reviewer: This would would be possible during the SNS swap already, right?
So maybe saying that this is only possible after the SNS canisters have been deployed would be more precise?

Once the asset canister has been deployed and permissions have been configured, the SNS decentralization swap can be started. You can learn more about launching an SNS [here](../launching/launch-summary-1proposal.md)

### Overwriting permissions

> No matter the state of permissions, using upgrade arguments the permissions lists can be set to new values. The upgrade argument is of type `(AssetCanisterArgs)`, where the contained types are:

Reviewer 2: ```suggestion
Upgrade arguments can be used by the permissions lists to set new values, regardless of the current state of the permissions.

The upgrade argument is of type `(AssetCanisterArgs)`, where the contained types are:

```
